### DISCUSSION TITLE
Using the custom cooldown utility for custom buffs

### DISCUSSION THREAD

**Discussion by Aelto** [Aelto/tw3-shared-utils#3] [General]
Discussion about the custom cooldown utility and how it could be used to implement new custom buffs in an easy way.

**Comment by Johannes5** [comment]
I'm simply copy-pasting from our RPO discussion:

```
"Regarding your fake-cooldown class - How can those buffs get assigned a title and an icon?

--------
Part of the reason why I was asking about creating new buffs is a mod idea I would title "Steady Presence":
One gripe I have with the game is that the crossbow doesn't deal enough damage.
And manual aiming is a fun mechanic, but it isn't encouraged.

Wouldn't it be great if successful aimed shots dealt significantly higher damage?
In addition to non-aimed shots also getting more powerful.

This could be tied to the skills Steady Shot and Steady Aim.
I also wanted to tie it to a new buff you get from meditation.
And I also wanted to do something with Sustained Glyphs for Yrden.

But coming to think of it:
Bonus Perks for Steady Shot and Steady Aim would be a great addition to RPO.

Don't you think?"
```

**Comment by Aelto** [comment]
> Regarding your fake-cooldown class - How can those buffs get assigned a title and an icon?
To assign a title and an icon to the custom buffs, you can set the values you need to the [`title` and `icon` properties](https://github.com/Aelto/tw3-shared-utils/blob/20b98d1b8534ca9785cebf274d17adee0ae1639f/mod_sharedutils_customcooldowns/content/scripts/local/sharedutils/custom-cooldowns/types.ws#L30-L32)

If you wish to see how you would use this sharedutils module, you can look at the [Consumables on cooldown](https://github.com/Aelto/tw3-consumables-on-cooldowns) mod i made, it contains a [single local file](https://github.com/Aelto/tw3-consumables-on-cooldowns/blob/master/modConsumablesOnCooldowns/content/scripts/local/consumables-on-cooldowns/main.ws) with everything you'd need to implement what you want:
 - define the custom buff, its duration, title and icon
 - verify if the player already has the buff
 - optionally do something when the buff expires
